src/Plugins/SystemCPlugin/Peripheral/SystemCPeripheral.cs

```diff
@@ -400,8 +484,14 @@ private void BackwardConnectionLoop()
                         }
                         break;
                     case RenodeAction.Write:
+                        bool writeToSharedMem = false;
                         if(message.IsSystemBusConnection())
                         {
+                            var targetMem = sysbus.FindMemory(message.Address);
+                            if (targetMem != null)
+                            {
+                                writeToSharedMem = targetMem.Peripheral.UsingSharedMemory;
```

Comment:
@ky-macpherson It seems MappedMemory doesn't have `UsingSharedMemory` member in mainline Renode; this fork also doesn't seem to have it. I'm unable to build and test the change because of that. Is it a feature from a different fork?

Reply:
My bad, I missed that this change is dependent on https://github.com/renode/renode-infrastructure/pull/114 - we'll look into that one ASAP.